modules/common/service/service.go

```diff
@@ -176,3 +177,38 @@ func GetServicesListWithLabel(
 
 	return serviceList, nil
 }
+
+// GetServiceWithName - Get service with name in namespace
+func GetServiceWithName(
+	ctx context.Context,
+	h *helper.Helper,
+	name string,
+	namespace string,
+) (*corev1.Service, error) {
+
+	service := &corev1.Service{}
+
+	err := h.GetClient().Get(ctx, types.NamespacedName{Name: name, Namespace: namespace}, service)
```

Comment:
Just one question.  You mentioned in the other PR (https://github.com/openstack-k8s-operators/lib-common/pull/158) that you wanted to use KClient to not be restricted to cache-only pods in watched namespaces.  Do we have any interest in doing that here too?